deploy/deploy-scaledobject.yaml

```
@@ -0,0 +1,22 @@
+# Template scaled-object for using KEDA external scaler for Azure Cosmos DB.
+
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: <scaledobject-name>
+  namespace: default
+spec:
+  pollingInterval: 20
+  scaleTargetRef:
+    name: <application-deployment-name>
+  triggers:
+    - type: external
+      metadata:
+        scalerAddress: keda-cosmosdb-scaler.keda:4050
+        connection: <connection-string-of-monitored-container-account>
+        databaseId: <database-id>
+        containerId: <container-id>
+        leaseConnection: <connection-string-of-lease-container-account>
+        leaseDatabaseId: <lease-database-id>
+        leaseContainerId: <lease-container-id>
+        processorName: <processor-name>
```

```suggestion
      metadata:
        scalerAddress: keda-cosmosdb-scaler.keda:4050
        connection: <connection-string-of-monitored-container-account>
        databaseId: <database-id>
        containerId: <container-id>
        lease:
          connection: <connection-string-of-lease-container-account>
          databaseId: <lease-database-id>
          containerId: <lease-container-id>
        processorName: <processor-name>
```

We could even do this already for database & container, but we can wait

---

I am not sure about this. For example, `lease` is not the right word here as we are not talking about a single lease; the lease container will be holding several leases, may be even for multiple monitored containers. What I am interested in is lease container, so the metadata representation might be `leaseContainer: { id: "", databaseId: "", connection: ""}`. The `processorName` is name of lease container change-feed processor, so it's associated with lease container but it cannot be part of lease container definition. That makes it difficult to choose whether to place it outside or inside the `lease` object. I tried taking some reference from [WebJobs extension](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/main/src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttribute.cs) where lease and monitored container properties are defined at same level. To me, the current placement of properties looks good enough.

---

@tomkerkhove - Do you think the following metadata would improve upon the existing one? If so, I can work on it in this or next week.

```
metadata:
  scalerAddress: keda-cosmosdb-scaler.keda:4050
  container:
    containerId: <container-id>
    databaseId: <database-id>
    connection: <connection>
  leaseContainer:
    id: <lease-container-id>
    databaseId: <lease-database-id>
    connection: <lease-connection>
  processorName: <processor-name>
  ```

---

I  think that would be lovely!